cmd/shutdown.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+const (
+	shutdownTimeout = 10 * time.Second
+)
+
+//GracefulShutdown implements releasing all resouces it got from system, finish all request handling and return responses when service stopping.
+func gracefulShutdown(done chan<- bool) error {
+	sigs := make(chan os.Signal)
+
+	timeout := make(chan bool)
+
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	sig := <-sigs
+	log.Println("Recieved", sig, "signal")
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	shutdownTime := time.Now().Add(shutdownTimeout)
```

Comment:
You can use context.WithTimeout instead of time variable

Reply:
fixed